crates/state-asm/src/effects.rs

```diff
@@ -0,0 +1,98 @@
+use crate::StateRead as StateReadOp;
+use bitflags::bitflags;
+use essential_constraint_asm::{Access, Constraint as ConstraintOp};
+
+/// Flags to indicate the effects of state read operations.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Effects(u8);
+
+bitflags! {
+    impl Effects: u8 {
+        /// Flag for [´StateReadOp::KeyRange´]
+        const Range = 1 << 0;
+        /// Flag for [´StateReadOp::KeyRangeExtern´]
+        const RangeExtern = 1 << 1;
```

Comment:
```suggestion
        /// Flag for [´StateReadOp::KeyRange´]
        const KeyRange = 1 << 0;
        /// Flag for [´StateReadOp::KeyRangeExtern´]
        const KeyRangeExtern = 1 << 1;
```
maybe better to match the original ops?